src/main/java/org/n52/wacodis/javaps/algorithms/ForestVitalityChangeAlgorithm.java

```diff
@@ -0,0 +1,173 @@
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.esa.snap.core.dataio.ProductIO;
+import org.esa.snap.core.datamodel.Product;
+import org.n52.javaps.algorithm.annotation.Algorithm;
+import org.n52.javaps.algorithm.annotation.ComplexOutput;
+import org.n52.javaps.algorithm.annotation.Execute;
+import org.n52.javaps.algorithm.annotation.LiteralInput;
+import org.n52.javaps.io.GenericFileData;
+import org.n52.wacodis.javaps.WacodisProcessingException;
+import org.n52.wacodis.javaps.command.AbstractCommandValue;
+import org.n52.wacodis.javaps.io.data.binding.complex.GeotiffFileDataBinding;
+import org.n52.wacodis.javaps.io.data.binding.complex.ProductMetadataBinding;
+import org.n52.wacodis.javaps.io.http.SentinelFileDownloader;
+import org.n52.wacodis.javaps.io.metadata.ProductMetadata;
+import org.n52.wacodis.javaps.io.metadata.ProductMetadataCreator;
+import org.n52.wacodis.javaps.io.metadata.SentinelProductMetadataCreator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ *
+ * @author <a href="mailto:s.drost@52north.org">Sebastian Drost</a>
+ */
+@Algorithm(
+        identifier = "de.hsbo.wacodis.forest_vitality_change",
+        title = "Forest Vitality Change",
+        abstrakt = "Perform a Forest Vitality Change calculation.",
+        version = "0.0.1",
+        storeSupported = true,
+        statusSupported = true)
+public class ForestVitalityChangeAlgorithm extends AbstractAlgorithm {
+
+    private static final String TIFF_EXTENSION = ".tif";
+    private static final String RESULTNAMEPREFIX = "forest-vitality-change_result";
+    private static final String TOOL_CONFIG = "forest-vitality-change.yml";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ForestVitalityChangeAlgorithm.class);
+
+    @Autowired
+    private SentinelFileDownloader sentinelDownloader;
+
+    private String opticalImagesSourceType;
+    private String opticalImagesSource, opticalImagesSource2;
+    private ProductMetadata productMetadata;
+    private Product sentinelProduct, sentinelProduct2;
+
+    @LiteralInput(
+            identifier = "OPTICAL_IMAGES_TYPE",
+            title = "Optical images source type",
+            abstrakt = "The type of the source for the optical images",
+            minOccurs = 1,
+            maxOccurs = 1,
+            defaultValue = "Sentinel-2",
+            allowedValues = {"Sentinel-2", "Aerial_Image"})
+    public void setOpticalImagesSourceType(String value) {
+        this.opticalImagesSourceType = value;
+
+    }
+
+    @LiteralInput(
+            identifier = "OPTICAL_IMAGES_SOURCES",
+            title = "Optical images sources",
+            abstrakt = "Sources for the optical images",
+            minOccurs = 1,
+            maxOccurs = 1)
+    public void setOpticalImagesSources(String value) {
+        this.opticalImagesSource = value;
+    }
```

Comment:
Please add a more meaningful description of the two optical image source inputs inside the @LiteralInput annotation in order to distinct these two inputs.